File: reverse_integer.py
class Solution:
    def reverse(self, x: int) -> int:
        reversedStringNum  =''.join(list(str(x))[len(list(str(x)))::-1])

        if reversedStringNum[-1] == '-':
            reversedInt = int(reversedStringNum[-1]+ reversedStringNum[0:-1])
            if (reversedInt > -2147483647) :
                return reversedInt
            else :
                return 0
        else :
            reversedInt = int(reversedStringNum)
            if (reversedInt < 2147483647) :
                return reversedInt
            else :
                return 0


        

File: reverse_integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    var numArrayReversed = x.toString().split('').reverse()
    var solution 
    if (numArrayReversed[numArrayReversed.length-1] === '-'){
        solution = -Math.abs(parseInt(numArrayReversed.join('')));
    } else{
        solution = parseInt(numArrayReversed.join(''));
    }
    if ((-2)**31 <= solution && solution <= 2**31 -1){
        return solution
    }else{
        return 0
    }
};